package/boot/arm-trusted-firmware-mvebu/patches-mox-boot-builder/002-no-gold.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/wtmi/Makefile b/wtmi/Makefile
+index 75754dc..0c6238f 100644
+--- a/wtmi/Makefile
++++ b/wtmi/Makefile
+@@ -36,7 +36,7 @@ LDSCRIPT = wtmi.ld
+ INCLUDE  = -I.
+ 
+ ifeq ($(LTO), 1)
+-	LTO_FLAGS = -flto -flto-partition=none -Wl,-fuse-ld=gold
++	LTO_FLAGS = -flto -flto-partition=none
```

Comment:
Why it is necessary to remove it? Can you please add commit description?

Reply:
Updated the commit description: Add a patch to disable linking with gold, since the ARM toolchain doesn't ship gold.

Reply:
Interesting, because ``mox-boot-builder`` is compiled successfully for ARM platforms even without your patch. This means that you do not somehow compile it with your environment and without providing details what is going on, you created and upstreamed to OpenWrt patch, which suits your environment. Would it be possible to share on which error you end up w/o forcing it on everyone, who is using mox-boot-builder? So it will be possible to get rid of this patch,  which should not be accepted here in the first place not even because of formal things.

Reply:
Now you're jumping to conclusions.

This is the error I get without this patch:
```
  AS       startup_asm.S
  CC       main.c
  LD       wtmi_app.elf
collect2: fatal error: cannot find 'ld'
compilation terminated.
           FAILED, trying with -lgcc
  LD -lgcc wtmi_app.elf
collect2: fatal error: cannot find 'ld'
compilation terminated.
make[5]: *** [Makefile:95: wtmi_app.elf] Error 1
```

Which matches the error I get when trying to reproduce this with the used toolchain:
```
echo 'int main() {return 0;}' | ./gcc-arm-10.2-2020.11-x86_64-arm-none-eabi/bin/arm-none-eabi-gcc -x c --specs=nosys.specs -Wl,-fuse-ld=gold -
collect2: fatal error: cannot find 'ld'
compilation terminated.
```

Because there is no gold linker bundled: `find gcc-arm-10.2-2020.11-x86_64-arm-none-eabi|grep gold`.

If I instead use `-Wl,-fuse-ld=ld.bfd` it succeeds, but that's because *now* my environment leaks into the build: debian ships `/usr/bin/ld.gold`

So I guess you have a `gold` binary somewhere in your path?

Reply:
Well, my question was why this patch is required now as it was not necessary to have it in the past, and I verified that it was working before.